src/api/app/controllers/webui/session_controller.rb

```diff
@@ -1,47 +1,48 @@
 class Webui::SessionController < Webui::WebuiController
   before_action :kerberos_auth, only: [:new]
 
+  before_action :session_creator, only: [:create]
+  before_action :authenticate, only: [:create]
+  before_action :check_user_active, only: [:create]
+
   skip_before_action :check_anonymous, only: [:create]
 
   def new
     switch_to_webui2
   end
 
   def create
-    user = User.find_with_credentials(params[:username], params[:password])
-
-    unless user
-      RabbitmqBus.send_to_bus('metrics', 'login,access_point=webui,failure=unauthenticated value=1')
-      redirect_to(new_session_path, error: 'Authentication failed')
-      return
-    end
-
-    unless user.is_active?
-      RabbitmqBus.send_to_bus('metrics', 'login,access_point=webui,failure=disabled value=1')
-      redirect_to(root_path, error: 'Your account is disabled. Please contact the administrator for details.')
-      return
-    end
-
-    User.session = user
-    session[:login] = user.login
-    Rails.logger.debug "Authenticated as user '#{user.login}'"
-    RabbitmqBus.send_to_bus('metrics', 'login,access_point=webui value=1')
-
+    User.session = @session_creator.user
+    session[:login] = @session_creator.user.login
+    SessionControllerNotificationJob.perform_later('login,access_point=webui value=1')
     redirect_on_login
   end
 
   def destroy
-    Rails.logger.info "Logging out: #{session[:login]}"
-
     reset_session
-    RabbitmqBus.send_to_bus('metrics', 'logout,access_point=webui value=1')
+    SessionControllerNotificationJob.perform_later('logout,access_point=webui value=1')
     User.session = nil
-
     redirect_on_logout
   end
 
   private
 
+  def session_creator
+    @session_creator = SessionControllerService::SessionCreator.new(params.slice(:username, :password))
+  end
+
+  def check_user_active
```

Comment:
This sounds like an extension to `SessionControllerService.valid?`

Reply:
can be, I will look on that

Reply:
not sure exactly what you mean, could you elaborate?